src/objects/objects.cc

```diff
@@ -5550,6 +5554,10 @@ Handle<Object> JSPromise::Reject(Handle<JSPromise> promise,
                                  kPromiseRejectWithNoHandler);
   }
 
+  if (RecordReplayShouldCallOnPromiseHook() && reason->IsJSReceiver()) {
+    AddPromiseDependencyGraphAdoption(isolate, Handle<Object>::cast(promise), reason);
+  }
```

Comment:
Is this needed? I'm not sure if my initial conclusion that it would be holds true 😅

Reply:
Yes it is. Rejections can also nest promises. Example:

```js
var p = Promise.resolve().then(() => new Promise(r => setTimeout(r,100), console.log("A")));
Promise.reject(p).finally(() => console.log("B"));
```

That being said, we have no tests for this yet, but I'll keep it in assuming that it'll work fine™.